Source/WebCore/dom/Document.h

```diff
@@ -1859,11 +1858,9 @@ class Document
     void setFragmentDirective(const String& fragmentDirective) { m_fragmentDirective = fragmentDirective; }
     const String& fragmentDirective() const { return m_fragmentDirective; }
 
-    void prepareCanvasesForDisplayIfNeeded();
-    void clearCanvasPreparation(HTMLCanvasElement&);
-    void canvasChanged(CanvasBase&, const std::optional<FloatRect>&) final;
-    void canvasResized(CanvasBase&) final { };
-    void canvasDestroyed(CanvasBase&) final;
+    void prepareCanvasesIfNeeded();
+    void updateCanvasPreparation(CanvasBase&);
```

Comment:
These names are not great; using "canvas preparation" as a noun makes it hard to know what they do. Preparation for what?

Maybe `updateCanvasForDisplayOrFlush` or something?